content/spin/serverless-ai-hello-world.md

```diff
@@ -0,0 +1,392 @@
+import json
+import re
+
+def handle_request(request):
+    try:
+        result = llm_infer("llama2-chat", "Can you tell me a joke abut cats")
+        return Response(200, {"content-type": "text/plain"}, bytes(result.text, "utf-8"))
+    except Exception as e:
+        return Response(500, {"content-type": "text/plain"}, bytes(f"Error: {str(e)}", "utf-8"))
+
+
+```
+ 
+{{ blockEnd }}
+ 
+{{ blockEnd }}
+ 
+### Building and Deploying Your Spin Application
+ 
+Now that you have written your first Serverless AI app, it’s time to build and deploy it. To build your app run the following commands from inside your app’s folder (where the `spin.toml` file is located)
+ 
+{{ tabs "sdk-type" }}
+ 
+{{ startTab "Rust"}}
+ 
+<!-- @selectiveCpy -->
+ 
+```bash
+$ spin build
+```
+ 
+{{ blockEnd }}
+ 
+{{ startTab "TypeScript" }}
+ 
+<!-- @selectiveCpy -->
+ 
+```bash
+
+$ npm install
+$ spin build
+
+```
+
+{{ blockEnd }}
+{{ startTab "Python" }}
+
+<!-- @selectiveCpy -->
+ 
+```bash
+
+$ spin build
+
+```
+
+{{ blockEnd }}
+ 
+Now that your app is built, there are three ways to test your Serverless AI app. One way to test the app is to run inferencing locally. This means running a LLM on your CPU. This is not as optimal compared to deploying to Fermyon’s Serverless AI which is performed by a high-powerful GPU in the cloud). To know more about this method, including downloading LLMs to your local machine, check out [the in-depth tutorial](https://developer.fermyon.com/spin/ai-sentiment-analysis-api-tutorial) on Building a Sentiment Analysis API using Serverless AI.
+ 
+Here are the two other methods for testing your app:
+ 
+1.  **Deploy the app to the Fermyon Cloud**
+ 
+You can deploy the app to the cloud by using the `spin deploy` command. In case you have not logged into your account before deploying your application, you need to grant access via a one-time token. Follow the instructions in the prompt to complete the auth process.
+ 
+Once you have logged in and the app is deployed, you will see a URL, upon successful deployment. The app is now deployed and can be accessed by anyone with the URL.
+ 
+```bash
+$ spin deploy
+
+>Uploading hello-world version 0.1.0+ra01f74e2...
+Deploying...
+Waiting for application to become ready...... ready
+Available Routes:
+hello-world: https://hello-world-XXXXXX.fermyon.app (wildcard)
+```
+ 
+The app’s manifest file reads the line `ai-models = ["llama2-chat"]` and uses that model in the cloud. For any changes to take effect in the app, it needs to be re-deployed to the cloud.
+ 
+2.  **Using the Cloud-GPU plugin to test locally**
+ 
+To avoid having to deploy the app for every change, you can use the [Cloud-GPU plugin](https://github.com/fermyon/spin-cloud-gpu) to deploy locally, with the LLM running in the cloud. While the app is hosted locally (running on `localhost`), every inferencing request is sent to the LLM that is running in the cloud. Follow the steps to use the `cloud-gpu` plugin
```

Comment:
```suggestion
To avoid having to deploy the app for every change, you can use the [Cloud-GPU plugin](https://github.com/fermyon/spin-cloud-gpu) to deploy locally, with the LLM running in the cloud. While the app is hosted locally (running on `localhost`), every inferencing request is sent to the LLM that is running in the cloud. Follow the steps to use the `cloud-gpu` plugin.
```